cmd/telegraf/telegraf.go

```diff
@@ -151,6 +155,40 @@ func reloadLoop(
 	}
 }
 
+// configDirFiles returns a slice made up of filepaths correspoding
+// to config files defined in any of the config directories.
+func configDirFiles() sliceFlags {
+	configs := make(sliceFlags, 0)
+	for _, dir := range fConfigDirs {
+		walkfn := func(thispath string, info os.FileInfo, _ error) error {
+			if info == nil {
+				log.Printf("W! Telegraf is not permitted to read %s", thispath)
+				return nil
+			}
+
+			if info.IsDir() {
+				// skip Kubernetes mounts, preventing loading the same config twice
+				if strings.HasPrefix(info.Name(), "..") {
+					return filepath.SkipDir
+				}
+				return nil
+			}
+
+			name := info.Name()
+			if len(name) < 6 || name[len(name)-5:] != ".conf" {
+				return nil
+			}
+
+			relPath := filepath.Join(dir, name)
+			configs = append(configs, relPath)
+			return nil
+		}
+
+		filepath.Walk(dir, walkfn)
```

Comment:
yeah I caught this one when I was writing it. I wasn't sure how it would be handled, since the `error` returned by the `walkfn` in `config.LoadDirectory()` is the return value for the `config.LoadDirectory()` itself, which makes sense -- an issue walking the directory would not be nice in a config load scenario. but here the only explicit err in `LoadDirectory()` (returned by `LoadConfig()`) is removed, so the walkn doesn't really return an error that we need to handle. but perhaps it's easiest and best practice to just do

```go
if err := filepath.Walk(dir, walkfn); err != nil {
     log.Printf("E! ...", ...)
}
```

Reply:
Or you could do something like `_ = filepath.Walk(dir, walkfn)`.

Reply:
We're not expecting an error here, but we shouldn't ignore it. Maybe we should have configDirFiles return sliceFlags and an error. Then we could return the error from filepath.Walk and the caller (reloadLoop) could handle it. reloadLoop would check the err and if not nil, print something and maybe skip appending the files.